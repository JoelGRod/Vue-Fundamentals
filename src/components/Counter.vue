<template>
  <h2> {{ customTitle }} </h2>
  <!-- Html Elements with Data Attributes for tests -->
  <p test-id="expressionCounter"> {{ counter }} <sup>2</sup> = {{ squareCounter }}  </p>
  <p test-id="counter"> {{ counter }}  </p>
  <div>
      <button test-id="increaseBtn" @click="increaseCounter" class="button">+1</button>
      <button test-id="decreaseBtn" @click="decreaseCounter" class="button">-1</button>
  </div>
</template>

<script>
export default {
    name: 'Counter',
    // Props: 1ยบ aprox
    // props: [ 'title' ],
    // Props: 2ยบ aprox
    props: {
        title: String,
        start: {
            type: Number,
            default: 1,
            required: false,
            validator: (value) => {
                return value >= 0
            }
        }
    },
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        getSquareValue() {
            // console.log("Method square value")
            return this.counter * this.counter
        },
        increaseCounter( event ) {
            this.counter++
        },
        decreaseCounter( event ) {
            this.counter > 0 ? this.counter-- : this.counter = 0
        }

    },
    computed: {
        squareCounter() {
            // console.log("Computed square counter")
            return this.counter * this.counter
        },
        customTitle() {
            return this.title ? this.title : "Counter"
        }
    }
}
</script>

<style>

    .button {
        background-color: #64BB87;
        border-radius: 5px;
        border: 1px solid white;
        color: white;
        cursor: pointer;
        margin: 0px 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    .button:hover {
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }

</style>