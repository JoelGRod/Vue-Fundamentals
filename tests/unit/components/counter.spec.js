import {shallowMount, mount} from '@vue/test-utils';
import Counter from '@/components/Counter';

describe('Counter Component', () => {
    test('It must match the snapshot', () => {

        // Arrange
        const wrapper = shallowMount( Counter );

        // Act

        // Assert
        expect( wrapper.html() ).toMatchSnapshot();
    })
})