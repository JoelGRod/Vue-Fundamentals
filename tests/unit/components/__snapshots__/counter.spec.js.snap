// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component It must match the snapshot 1`] = `
<h2>Counter</h2>
<p>1 <sup>2</sup> = 1</p>
<div><button class="button">+1</button><button class="button">-1</button></div>
`;
